### ISSUE TITLE
Why can open_end=True and/or open_begin=True lead to not finding a warping function?

### ISSUE THREAD

**Issue by tcwalther** [DynamicTimeWarping/dtw-python#4]
I noticed that in certain cases, setting `open_end=True` and/or `open_begin=True` can lead to not finding a warping function. I found this surprising, as intuitively, I would have expected the removal of one or several constraints (i.e., start and end have to match) to make it easier to find a warping function, not harder.

I've attached two files for an example time series to reproduce the proplem:

```python
>>> import numpy as np
>>> from dtw import *

>>> x = np.loadtxt("x.txt")
>>> y = np.loadtxt("y.txt")

# works fine
>>> _ = dtw(x, y, step_pattern="asymmetricP05")

# doesn't work
>>> _ = dtw(x, y, step_pattern="asymmetricP05", open_end=True)
ValueError: No warping path found compatible with the local constraints
```

Is this a bug, or am I missing something in my understanding of DTW?

[x.txt](https://github.com/DynamicTimeWarping/dtw-python/files/4241325/x.txt)
[y.txt](https://github.com/DynamicTimeWarping/dtw-python/files/4241326/y.txt)

**Comment by tonigi** [comment]
Uhm... yes it seems a bug. Works in R. Will investigate.

**Comment by tcwalther** [comment]
Thanks @tonigi!

**Comment by tonigi** [comment]
Fixed in af5a5271397a19337582149a57614087a0be393a and release 1.0.5 :) Thanks!

**Comment by tcwalther** [comment]
Absolutely amazing. I can confirm that the fix works.

Thanks @tonigi, that was really quick!
